### PR TITLE
refactor: rename DBErrors::RESCAN_REQUIRED to NEED_RESCAN

### PR DESCRIPTION
**Pull request by meshcollider**
Suggested here as a trivial follow-up: https://github.com/bitcoin/bitcoin/pull/23123#discussion_r719709924

Makes RESCAN_REQUIRED consistent with NEED_REWRITE

### CODE REVIEW

**Comment by achow101** [comment]
scripted diff?

**Comment by meshcollider** [comment]
Added scripted diff

File: src/wallet/walletdb.h [hunk 1]
```diff
@@ -49,7 +49,7 @@ enum class DBErrors
     TOO_NEW,
     LOAD_FAIL,
     NEED_REWRITE,
-    RESCAN_REQUIRED
+    NEED_RESCAN
```
Comment (jonatack) [inline src/wallet/walletdb.h:52]: if you have to retouch
```suggestion
    NEED_RESCAN,
```

**Review by jonatack — COMMENTED** [review]
ACK 8615507a4e510bc797b01c4bbe45dc1f91e0bfa0

**Comment by achow101** [comment]
ACK 8615507a4e510bc797b01c4bbe45dc1f91e0bfa0

**Comment by fanquake** [comment]
This has been merged.

**Pull request closed without merging**